2017/11/29/systemd-primer-from-the-trenches.html.md

```diff
@@ -0,0 +1,514 @@
+add-wants              daemon-reexec          emergency              help                   isolate                list-jobs              poweroff               reload-or-try-restart  set-property           stop                   
+cancel                 daemon-reload          enable                 hibernate              is-system-running      list-sockets           preset                 rescue                 show                   suspend                
+cat                    default                exit                   hybrid-sleep           kexec                  list-timers            reboot                 reset-failed           show-environment       switch-root            
+condreload             delete                 force-reload           is-active              kill                   list-unit-files        reenable               restart                snapshot               try-restart            
+condrestart            disable                get-default            is-enabled             link                   list-units             reload                 set-default            start                  unmask 
+```
+
+Lets start at the top and work down;
+
+1.
+
+```systemctl``` 
+
+```service```
+
+Used in conjunction with ABRT can show you some great debug info and runtime metadata, categorised in their respective groupings of loaded, 
+active, running and a description of the unit.
+
+2.
+```systemctl status```
+
+```service --status-all```
+or
+```initctl list```. If jobs are started in upstart on Ubuntu systems you will need to run this as well as `service --status-all`
+
+Check all system services status. Normally during a server update I will run this and output it to a file. 
+When the server reboots I can run it again and diff this file to ensure all things started. The output is great. It shows me the 
+PID path and potentially the arguments which were run for the process or service. Saves me `ps`'ing the process.
+
+Note, each distro deals differently with `service --status-all` output. 
+
+3.
+```systemctl status serviceName -l```
+
+Good flags: is-active or -a 
+
+```service serviceName status```
+
+As it suggests, show me the status and information related to the service Unit file. Other good info includes is the service enabled or
+is chkconfig on, uptime, PID and CGroup info, plus any other information associated with the service. 
+
+Tip: 
+* use the -l flag
+* To view more than one service group services in {} braces, i.e.  {httpd,mysql,postfix}
+
+4.
+```systemctl enable NameofService```
+
+Flags: disable
+
+```chkconfig on ServiceName```
+
+Flags: on or off
+
+You might find on some distro's that chkconfig is still present. It doesn't do what you think it does with systemd systems.
+
+Tip:
+* No tip on this, just remember, `enable` vs `chckconfig` and you will be swell.
+
+5.
+```systemctl start/stop/restart httpd```
+
+Good flags: reload-or-restart
```

Comment:
reload-or-restart isn't really a flag, it's a first-level command like start or stop, right? Might be better to just show an example of that. I had to look it up in the manpage since I hadn't seen reload-or-restart before and wasn't sure it was right.

Reply:
being fairly generic to keep it simple, but yes should adjust where flag is a flag and cmd is a cmd